### ISSUE TITLE
[bug] Using swagger for API docs

### ISSUE THREAD

**Issue by pattisdr** [django-json-api/django-rest-framework-json-api#57]
<h1> Problem </h1>

We use Swagger for API documentation. https://github.com/marcgibbons/django-rest-swagger.   There are two problems when trying to use this package in conjunction with Swagger.

1) Since the package enforces the media-type  'application/vnd.api+json', it will cause Swagger to throw an error.  `{"errors":{"detail":"Could not satisfy the request Accept header."}}`  I've worked around this in the past using custom content negotiation.

![screen shot 2015-08-10 at 4 08 12 pm](https://cloud.githubusercontent.com/assets/9755598/9182845/bd063bb6-3f79-11e5-9d03-0255dc53fa3a.png)

2)  Provided the first issue is fixed, Swagger hangs with a fetching resource list message.
![screen shot 2015-08-10 at 4 12 21 pm](https://cloud.githubusercontent.com/assets/9755598/9182945/56c1b2a8-3f7a-11e5-913b-c743b6940ea9.png)

<h1> Potential Solutions </h1>
I haven't dug into this much, but one thing I've noticed is that this package is looking for `views` to have an action.  I assume this means even SwaggerViews need an action.  If I add an action to the SwaggerView, it still hangs.

**Comment by jsenecal** [comment]
For problem 1, see http://jsonapi.org/format/#content-negotiation-clients

**Comment by brianjgeiger** [comment]
After chatting with @pattisdr, it looks like problem 2 is, essentially, https://github.com/marcgibbons/django-rest-swagger/issues/13. I expect it's going to have to be fixed on the django-rest-swagger end based on what I've seen so far, but if anyone sees a way around it on this end, it would be worth knowing.

**Comment by jsenecal** [comment]
@pattisdr can we close this?

**Comment by jerel** [comment]
Let's leave it open until there's some sort of resolution on the Swagger project (ideally) or we figure out a workaround here.

**Comment by pattisdr** [comment]
The problem was caused by some changes I had made to the renderers.

I modified renderers.py because my `meta` information was showing up twice.  
```
    if data != json_api_data:
            render_data['data'] = json_api_data
```
![image](https://cloud.githubusercontent.com/assets/9755598/9305555/45759370-44c0-11e5-9cdc-016cf66cdb6d.png)

So when I tried to go to Swagger docs (http://localhost:8000/v2/docs/), `render_data` was just an OrderedDict(), with nothing inside. This caused Swagger to hang with a `fetching resource list` and the error below because there was no `response.apis` returned.  
![screen shot 2015-08-17 at 9 08 45 am](https://cloud.githubusercontent.com/assets/9755598/9305424/6197ed74-44bf-11e5-84d7-e8b75bdc3b03.png)

Adding the bottom lines then, both prevented my meta information from showing up twice, but still placed data in `render_data` for the Swagger docs.
![screen shot 2015-08-17 at 9 27 27 am](https://cloud.githubusercontent.com/assets/9755598/9305771/f03fa6aa-44c1-11e5-9408-0a6207d70130.png)
